src/runtime_src/core/pcie/tools/xbmgmt/cmd_hotplug.cpp

```diff
@@ -0,0 +1,197 @@
+    auto mDev = pcidev::get_dev(index, false);
+    removeDevice(mDev);
+    
+    return 0;
+}
+
+static int removeDevice(std::shared_ptr<pcidev::pci_device> dev)
+{
+    std::string sysfs_path = dev->get_sysfs_path("", "remove");
+
+    if (sysfs_path.empty()) {
+        return -ENOENT;
+    }
+
+    std::ofstream ofile(sysfs_path);
+
+    if (!ofile.is_open()) {
+        std::cout << "Failed to open " << sysfs_path << ":" << strerror(errno) << std::endl;
+        return -ENOENT;
+    }
+
+    /* "echo 1 > /sys/bus/pci/<EndPoint>/remove" to trigger hot remove of the device */
+    ofile << 1;
+    ofile.flush();
+    if (!ofile.good()) {
+        std::cout << "Failed to write " << sysfs_path << ":"  << strerror(errno) << std::endl;
+        ofile.close();
+        return -EINVAL;
+    }
+
+    ofile.close();
+
+    return 0;
+}
+
+
+const char *subCmdHpRescanDesc = "Perform hot rescan on the device";
+const char *subCmdHpRescanUsage = "(no options supported)";
+
+int hpRescanHandler(int argc, char *argv[])
+{
+    sudoOrDie();
+
+    if (argc != 1)
+        return -EINVAL;
+
+    const std::string sysfs_path = findRootPort() + "/rescan";
+
+    if (sysfs_path.empty()) {
+        return -ENOENT;
+    }
+
+    std::ofstream ofile(sysfs_path);    
+    
+    if (!ofile.is_open()) {
+        std::cout << "Failed to open " << sysfs_path << ":" << strerror(errno) << std::endl;
+        return -ENOENT;
+    }
+   
+    /* "echo 1 > /sys/bus/pci/<Root Port>/rescan" to trigger the rescan for hot plug devices */ 
+    ofile << 1;
+    ofile.flush();
+    if (!ofile.good()) {
+        std::cout << "Failed to write " << sysfs_path << ":"  << strerror(errno) << std::endl;
+        ofile.close();
+        return -EINVAL;
+    }
+
+    ofile.close();
+
+    return 0;
+}
+
+static std::string findRootPort(void)
+{
+    std::string rootPortPath = "";
+    path fPath(SYSFS_PATH);
+    
+    for (auto file = directory_iterator(fPath); file != directory_iterator(); file++)
+    {
+        rootPortPath = file->path().string();
+        for (auto jfile = directory_iterator(rootPortPath); jfile != directory_iterator(); jfile++)
+        {
+            std::string dirName(jfile->path().string());
+            if (is_directory(dirName))
+            {
+                std::string vendor_id, device_id;
+                std::string vendorPath = dirName + "/vendor";
+                
+                if (boost::filesystem::exists(vendorPath))
+                {
+                    std::ifstream file(vendorPath);
+                    std::getline(file, vendor_id);
+                }
+                    
+                std::string devicePath = dirName + "/device";
+                if (boost::filesystem::exists(devicePath))
+                {
+                    std::ifstream file(devicePath);
+                    std::getline(file, device_id);
+                }
+
+                if (!strcmp(vendor_id.c_str(), XILINX_VENDOR) && !strcmp(device_id.c_str(), XILINX_US))
```

Comment:
Do not go back to C to operate on C++ objects
`if (!vendor_id.compare(XILINX_VENDOR) && !device_id.compare(XILINX_US))`

Reply:
Thanks for this suggestion. Incorporate the same.